aws-logs-loggroup/src/main/java/software/amazon/logs/loggroup/UpdateHandler.java

```diff
@@ -69,6 +80,44 @@ private void putRetentionPolicy(final AmazonWebServicesClientProxy proxy,
         logger.log(retentionPolicyMessage);
     }
 
+    private void disassociateKmsKey(final AmazonWebServicesClientProxy proxy,
+                                    final ResourceHandlerRequest<ResourceModel> request,
+                                    final Logger logger) {
+        final ResourceModel model = request.getDesiredResourceState();
+        final DisassociateKmsKeyRequest disassociateKmsKeyRequest =
+                Translator.translateToDisassociateKmsKeyRequest(model);
+        try {
+            proxy.injectCredentialsAndInvokeV2(disassociateKmsKeyRequest,
+                    ClientBuilder.getClient()::disassociateKmsKey);
+        } catch (final ResourceNotFoundException e) {
+            throwNotFoundException(model);
+        }
+
+        final String kmsKeyMessage =
+                String.format("%s [%s] successfully disassociated kms key.",
+                        ResourceModel.TYPE_NAME, model.getLogGroupName());
+        logger.log(kmsKeyMessage);
+    }
+
+    private void associateKmsKey(final AmazonWebServicesClientProxy proxy,
+                                 final ResourceHandlerRequest<ResourceModel> request,
+                                 final Logger logger) {
+        final ResourceModel model = request.getDesiredResourceState();
+        final AssociateKmsKeyRequest associateKmsKeyRequest =
+                Translator.translateToAssociateKmsKeyRequest(model);
+        try {
+            proxy.injectCredentialsAndInvokeV2(associateKmsKeyRequest,
+                    ClientBuilder.getClient()::associateKmsKey);
+        } catch (final ResourceNotFoundException e) {
```

Comment:
[There are more exceptions that can be thrown](https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_AssociateKmsKey.html#API_AssociateKmsKey_Errors)